File: apps/src/chat/frontend/js/packets/websocket/JoinPacket.js
export default class JoinPacket {

    constructor(username = ""){
        this.username = username;
    }

    toData(){
        return {
            username: this.username,
            type: "join"
        }
    }

    static fromData(data){
        return new JoinPacket(data["username"]);
    }

    getUsername(){
        return this.username;
    }
}

File: apps/src/chat/backend/packets/websocket/join_packet.py
from ..packet import Packet

class JoinPacket(Packet):

    def __init__(self, username: str):
        self._username = username

    def to_data(self):
        return {
            "username": self._username,
            "type": "join"
        }

    @staticmethod
    def from_data(data):
        return JoinPacket(
            data["username"]
        )